tools/src/launcher/Main.hs

```diff
@@ -100,6 +100,20 @@ type M a = (HasConfigurations, HasCompileInfo) => Log.LoggerNameBox IO a
 
 data Executable = EWallet | ENode | EUpdater
 
+-- Node, its args, node log
+data NodeData = NodeData
```

Comment:
I'd write a comment that this type can also carry wallet data (wallet is also a node, but it's not obvious whether `NodeData` corresponds only to wallet or to either node or wallet).